runtime/src/accounts_db.rs

```diff
@@ -809,6 +809,26 @@ impl<'a> LoadedAccountAccessor<'a> {
             }
         }
     }
+
+    fn account_matches_owners(&self, owners: &[&Pubkey]) -> Option<bool> {
```

Comment:
I think the Option return value is more confusing than it could be. I suggest an enum:
matches, does not match, unable to determine (for overflow case). Or, maybe even a result since unable to determine is an error (which should ideally never happen!?!?)

Reply:
Would you like a similar change for the `append_vec.rs` API? 
I had modeled this mostly on `load()` API. And, a lot of internal APIs currently return an `Option`. I do agree that a `Result` with specific errors will be cleaner approach.

Reply:
Changed all the new APIs to return `Result`